<?xml version="1.0" encoding="UTF-8"?>

<!--===================GNSS=================-->
<launch>
  <arg name="node_name" value="ublox_gps"/>
	    <arg name="param_file_name" value="zed-f9p"/>
	    <arg name="output" value="screen"/>
	    <arg name="respawn" value="true"/>
	    <arg name="respawn_delay" value="30"/>
	    <arg name="clear_params" value="false"/>

  <node pkg="ublox_gps" type="ublox_gps" name="$(arg node_name)"
        output="$(arg output)" 
        clear_params="$(arg clear_params)"
        respawn="true" 
        respawn_delay="$(arg respawn_delay)">
    <rosparam command="load" 
              file="$(find ublox_gps)/config/$(arg param_file_name).yaml" />
 </node>

<!--=====================GPS_W_HEADER=====================-->
  <node pkg="local_pkg" type="gps_w_header.py" name="gps_w_header" respawn="true"/>

<!--=====================ENCODER=====================-->
  <node pkg="local_pkg" type="encoder_parser.py" name="encoder" respawn="true"/>

<!--=====================AHRS=====================-->
  <node pkg="local_pkg" type="ahrs_parser.py" name="ahrs_raw" respawn="true" output="screen"/>

<!--=====================message_filter=====================-->
  <node pkg="local_pkg" type="local_sensors_w_timesync.py" name="message_synchronizer_local" respawn="true" output="screen"/>

<!--================LOCAL=======================-->
  <node pkg="local_pkg" type="local.py" name="Localization" respawn="true" output="screen"/>

</launch>
